backend/src/test/java/de/neuefische/booklibrary/backend/BookServiceTest.java

```diff
@@ -67,7 +67,7 @@ void addNewBookWithoutId_returnBookWithId() {
     @Test
     void addNewBookWithId_returnBook() {
         //given
-        Book book = new Book("id1", "Java-Script", "P. Ackermann", "978-3-8362-8629-9", BookState.NOT_AVAILABLE);
+        Book book = new Book("id1", null, "Java-Script", "P. Ackermann", "978-3-8362-8629-9", BookState.NOT_AVAILABLE);
         Book saveBook = book.withId("id1").withTitle("Java-Script").withAuthor("P. Ackermann").withIsbn("978-3-8362-8629-9").withBookState(BookState.NOT_AVAILABLE);
         when(bookRepository.save(saveBook)).thenReturn(saveBook);
```

Comment:
You are not creating IDs in your service, right? Consider generating UUIDs and then save it in the database